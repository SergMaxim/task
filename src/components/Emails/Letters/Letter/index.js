import React from 'react';
import moment from 'moment';
import { bool, number, string } from 'prop-types';

import Styles from './styles.scss';

const Letter = (props) => {
    let style = null;

    if (props.star) {
        style = {
            backgroundPosition: 'top center'
        };
    }

    return (
        <section className = { Styles.letter }>
            <span className = { Styles.star } style = { style } />
            <div className = { Styles.name }> {props.name}</div>
            <div className = { Styles.theme }> {props.theme}</div>

            <time>
                {moment
                    .unix(props.time)
                    .locale('uk')
                    .format('D.M.YY HH:mm')}
            </time>
        </section>
    );
};

Letter.propTypes = {
    name:  string.isRequired,
    star:  bool.isRequired,
    theme: string.isRequired,
    time:  number.isRequired
};

export default Letter;
