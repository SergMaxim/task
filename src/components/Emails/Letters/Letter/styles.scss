.letter{
  position: relative;
  display: flex;
  /*justify-content: space-between;*/
  align-items:center;
  height: 40px;
  line-height: 40px;
  border-bottom: 1px solid #d8e3e7;
  cursor: pointer;
  padding-left: 10px;

  &:hover{
    background-color: #ffffff;
  }
  .star, .name, .theme{
    box-sizing: content-box;
    color: #404661;
    font-size: 14px;
  }
  .star{
    max-width: 10%;
    width: 20px;
    height: 20px;
    background-position: bottom center;
    background-size: 100% 200%;
    background-repeat: no-repeat;
    background-image: url("./assets/star.png");
  }
  .name{
    width: 30%;
    padding-left: 10px;
  }
  .theme{
    width: 50%;
  }
  time{
    padding-right: 20px;
    width: 20%;
    padding-left: 10px;
    text-align: right;
  }
}