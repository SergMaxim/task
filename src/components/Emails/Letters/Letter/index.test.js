import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import dom from 'react-test-renderer';

// Instruments
import Letter from './';

configure({ adapter: new Adapter() });

const message = 'message';
const props = {
    name:  'name',
    star:  true,
    theme: 'theme',
    time:  1513545795
};
const state = {
    comment: ''
};
const mutatedState = {
    comment: message
};

const result = mount(<Letter { ...props } />);
const renderTree = dom.create(<Letter { ...props } />).toJSON();

describe('Letter component', () => {
    test(`Letter should have 1 'section' element`, () => {
        expect(result.find('section')).toHaveLength(1);
    });

    test(`Letter should have 1 'span' element`, () => {
        expect(result.find('span')).toHaveLength(1);
    });

    test(`Letter should have 1 'time' element`, () => {
        expect(result.find('time')).toHaveLength(1);
    });

    test(`Letter should div 2 'time' element`, () => {
        expect(result.find('div')).toHaveLength(2);
    });
    test('Letter component should correspond to its snapshot counterpart', () => {
        expect(renderTree).toMatchSnapshot();
    });
    test('Letter style should correspond to its snapshot counterpart', () => {
        expect(result.find('.star').props().style.backgroundPosition).toBe(
            'top center'
        );
    });
});
