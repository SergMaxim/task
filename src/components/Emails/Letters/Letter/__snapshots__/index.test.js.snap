// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter component Letter component should correspond to its snapshot counterpart 1`] = `
<section
  className="letter"
>
  <span
    className="star"
    style={
      Object {
        "backgroundPosition": "top center",
      }
    }
  />
  <div
    className="name"
  >
     
    name
  </div>
  <div
    className="theme"
  >
     
    theme
  </div>
  <time>
    17.12.17 23:23
  </time>
</section>
`;
